/*
https://www.hackerrank.com/challenges/java-exception-handling-try-catch/problem?h_r=internal-search
*/




import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
   
            try{
                Scanner sc = new Scanner(System.in);
                int a = sc.nextInt();
                int b = sc.nextInt();
                System.out.println(a/b);
                
            }
            catch(InputMismatchException ex){
                System.out.println("java.util.InputMismatchException");
            }
            catch(ArithmeticException ex){
                System.out.println("java.lang.ArithmeticException: / by zero");
            }
            catch(Exception ex){
                System.out.println("java.util.InputMismatchException");
            }
    
    }
}
